### PR TITLE
Strip cas

### PR DESCRIPTION
**Pull request by vgunning**
### PR checklist

- [x] Follows the [Developer Guide](https://sigma-dev.github.io/sigma/docs/developer-guide/).
- [x] Filed the PR against the correct branch?

### Change log

- Added a MetaImpl.Strip function to remove a key from the Metadata map.
- Updated Insert and Update Couchbase functions to strip any CAS value in the object's metadata.
- Consolidated all string type declarations to exported constants example: `ServiceType = "SERVICE"`
- Standardized all Update and Insert Couchbase functions to add the object type to the metadata.

### CODE REVIEW

File: model/model.go [hunk 1]
```diff
@@ -59,6 +59,17 @@ func (m *MetaImpl) Get(k string) (interface{}, bool) {
 	return v, ok
 }
 
+//Strip removes the key from the map if it exists
+func (m *MetaImpl) Strip(k string) {
+	if m.Metadata == nil {
+		m.Metadata = make(map[string]interface{})
+	}
+	_, ok := m.Metadata[k]
```
Comment (mikeynap) [inline model/model.go]: This is unneeded. no need to check if exists before calling delete.

**Comment by mikeynap** [comment]
Strip The Casbah

File: database/couchbase.go [hunk 2]
```diff
@@ -385,26 +383,35 @@ func (cb *Couchbase) InsertSite(m model.Site) (string, error) {
 
 func (cb *Couchbase) UpdatePlugin(m model.Plugin) error {
 	m.Set("Type", PluginType)
+	m.Strip("cas")
```
Comment (adamschaub) [inline database/couchbase.go]: Stripping cas in any of the "update" functions before the update happens will cause nothing to pass couchbase's cas check. Move `m.Delete("cas")` into `update` after we've fetched the cas from metadata.

File: database/couchbase.go [hunk 3]
```diff
@@ -364,6 +360,7 @@ func (cb *Couchbase) InsertBundle(m model.Bundle) (string, error) {
 
 func (cb *Couchbase) InsertServiceOperation(m model.ServiceOperation) (string, error) {
 	m.Set("Type", ServiceOperationType)
+	m.Strip("cas")
```
Comment (adamschaub) [inline database/couchbase.go]: If this is true for every update, modify the `insert` func to do the stripping. You'll need to pass in the model as a Metadater interface.

File: model/model.go [hunk 4]
```diff
@@ -59,6 +59,17 @@ func (m *MetaImpl) Get(k string) (interface{}, bool) {
 	return v, ok
 }
 
+//Strip removes the key from the map if it exists
+func (m *MetaImpl) Strip(k string) {
```
Comment (adamschaub) [inline model/model.go]: Unnecessary, just use `Delete`

**Pull request merged**